lightkurve/correctors/designmatrix.py

```diff
@@ -379,44 +429,396 @@ def __getitem__(self, key):
+        dm.X = dm.X[:, non_zero]
+        if dm.columns is not None:
+            dm.columns = list(np.asarray([['{}_{}'.format(c, idx) for c in dm.columns] for idx in range(len(row_indices) + 1)]).ravel())
+        dm.prior_mu = dm.prior_mu[non_zero]
+        dm.prior_sigma = dm.prior_sigma[non_zero]
+        return dm
+
+
+    def standardize(self, inplace=False):
+        """Returns a new `.SparseDesignMatrix` in which the columns have been
+        mean-subtracted and sigma-divided.
+
+        For each column in the matrix, this method will subtract the mean of
+        the column and divide by the column's standard deviation, i.e. it
+        will compute the column's so-called "standard scores" or "z-values".
+
+        This operation is useful because it will make the matrix easier to
+        visualize and makes fitted coefficients easier to interpret.
+
+        Notes:
+        * Standardizing a spline design matrix will break the splines.
+        * Columns with constant values (i.e. zero standard deviation) will be
+        left unchanged.
+
+        Returns
+        -------
+        `.SparseDesignMatrix`
+            A new design matrix with mean-subtracted & sigma-divided columns.
+        """
+        if inplace:
+            dm = self
+        else:
+            dm = self.copy()
+
+        idx, jdx, v = find(dm.X)
+        weights = dm.X.copy()
+        weights[dm.X != 0] = 1
+        mean = np.bincount(jdx, weights=v)/np.bincount(jdx)
+        std = np.asarray([((np.sum((v[jdx == i] - mean[i])**2) * (1/((jdx == i).sum() - 1))))**0.5 for i in np.unique(jdx)])
+        mean[std == 0] = 0
+        std[std == 0] = 1
+        white = (dm.X - vstack([lil_matrix(mean)] * dm.shape[0])).multiply(vstack([lil_matrix(1/std)] * dm.shape[0]))
+        dm.X = white.multiply(weights)
+        return dm
+
+    def pca(self, nterms=6, **kwargs):
+        """Returns a new `.SparseDesignMatrix` with a smaller number of regressors.
+
+        This method will use Principal Components Analysis (PCA) to reduce
+        the number of columns in the matrix.
+
+        Parameters
+        ----------
+        nterms : int
+            Number of columns in the new matrix.
+
+        Returns
+        -------
+        `.SparseDesignMatrix`
+            A new design matrix with PCA applied.
+        """
+        # nterms cannot be langer than the number of columns in the matrix
+        if nterms > self.shape[1]:
+            nterms = self.shape[1]
+        # We use `fbpca.pca` instead of `np.linalg.svd` because it is faster.
+        # Note that fbpca is randomized, and has n_iter=2 as default,
+        # we find this to be too few, and that n_iter=10 is still fast but
+        # produces more stable results.
+        from fbpca import pca  # local import because not used elsewhere
+        new_values, _, _ = pca(self.values, nterms, n_iter=10)
+        return SparseDesignMatrix(type(self.X)(new_values), name=self.name, **kwargs)
```

Comment:
A lot of code here is duplicated from `DesignMatrix.pca()`.

It would be better practice to call the parent method, e.g.:
```python
dense_dm = super().pca(nterms)
return SparseDesignMatrix(type(self.X)(dense_dm.X), name=self.name)
```

or perhaps

```python
return super().pca(nterms).to_sparse()
```

If that doesn't work, both methods could at least share the same helper function.

Also note: `SparseDesignMatrix.pca()` accepts `**kwargs` and `DesignMatrix.pca()` doesn't.  When overriding a method, changing the function signature should be avoided if possible.